packages/ui-tree-browser/src/TreeBrowser/TreeCollection/index.js

```diff
@@ -147,51 +153,128 @@ class TreeCollection extends Component {
   }
 
   get childCount() {
-    return this.collectionsCount + this.itemsCount
+    return (
+      this.collectionsCount +
+      this.itemsCount +
+      (this.props.beforeCollection ? 1 : 0) +
+      (this.props.afterCollection ? 1 : 0)
+    )
   }
 
   renderChildren() {
-    const { expanded, collections, items, name } = this.props
-
+    const {
+      expanded,
+      collections,
+      items,
+      name,
+      id,
+      beforeCollection,
+      afterCollection
+    } = this.props
+    const showList = this.childCount > 0 || beforeCollection || afterCollection
```

Comment:
If childCount is 0, then wouldn't beforeCollection and afterCollection be null (otherwise childCount above would have returned something greater than 0)?